server/sync_watch.go

```diff
@@ -51,13 +54,15 @@ func startSyncWatchProcess(server *Server) {
 		}
 		extensions[event] = env
 	}
-	responseChan := make(chan *gohan_sync.Event)
+	responseChans := make(map[string]chan *gohan_sync.Event)
 	stopChan := make(chan bool)
 	for _, path := range watch {
+		responseChans[path] = make(chan *gohan_sync.Event)
 		go func(path string) {
+			responseChan := responseChans[path]
 			defer util.LogFatalPanic(log)
 			for server.running {
-				lockKey := lockPath + "/watch"
+				lockKey := lockPath + "/watch" + path
```

Comment:
etcdv3's key names are simple strings and etcd doesn't handle key names as slash separated strings. So simple concatenating would be better I think.